src/libsystemd/sd-event/event-source.h

```diff
@@ -214,6 +214,10 @@ struct inotify_data {
          * the events locally if they can't be coalesced). */
         unsigned n_pending;
 
+        /* If this counter is non-zero, don't GC the inotify data object when it isn't needed anymore. This
+         * is useful to pin the object for a bit longer, even when it is otherwise not needed anymore. */
```

Comment:
This comment is an oxymoron: we prevent destruction because it is needed for *something*. I think this should say something like "dont' GC the inotify data object even when it not needed for …".